### PR TITLE
43 read support log file

### LINKED ISSUE #43: Add support to read log files of  SummaryWriter
The log of training is written by `SummaryWriter`. It is not possible to change file name and format, but we can add timestep as suffix, see [filename_suffix](https://docs.pytorch.org/docs/stable/tensorboard.html#torch.utils.tensorboard.writer.SummaryWriter). Also we can add a util function to read the log file using [tbparse](https://github.com/j3soon/tbparse).

### PR DESCRIPTION
**Pull request by rogerkuou**
fix #43 

- [x] add timestamp suffix
- [ ] add a logfile reader utility functin using [tbparse](https://github.com/j3soon/tbparse) -> there seems no need?
- [x] add example of reading logfiles in dailexample notebook

### CODE REVIEW

**Comment by rogerkuou** [comment]
Hi @SarahAlidoost , I added the timestamp to logfile suffix. However I found `tbparse` already provides very nice interface reading logs as DataFrames, [see these lines of usages](https://github.com/ESMValGroup/ClimaNet/blob/fc867aea517e0dd999012b386a64f6eb93cf82bf/tests/test_utils.py#L16-L19).
For now, I do not see the need to create a utility functions to read log files. Unless you thin the logfile should be read as a specific format? Or maybe I can add the use of [tbparse](https://github.com/j3soon/tbparse) to the example notebook?

**Comment by SarahAlidoost** [comment]
> Hi @SarahAlidoost , I added the timestamp to logfile suffix. However I found `tbparse` already provides very nice interface reading logs as DataFrames, [see these lines of usages](https://github.com/ESMValGroup/ClimaNet/blob/fc867aea517e0dd999012b386a64f6eb93cf82bf/tests/test_utils.py#L16-L19). For now, I do not see the need to create a utility functions to read log files. Unless you thin the logfile should be read as a specific format? Or maybe I can add the use of [tbparse](https://github.com/j3soon/tbparse) to the example notebook?

Hi @rogerkuou Thanks! There are a information in train.py and predict.py that are written by SummaryWriter, for example `Loss/best`. Can you please add an example to the notebook on how to read those information? This is useful when verbose=False.

**Comment by rogerkuou** [comment]
Hi @SarahAlidoost , I updated the variable names in train to solve the duplication of the training loss. Also added an example of reading log and plotted the history of training/validation loss.
Can you give this another look? Thanks!

**Comment by rogerkuou** [comment]
@SarahAlidoost and just solved the conflict. Feel free to review!

File: climanet/train.py
Comment (SarahAlidoost) [inline climanet/train.py:162]:
```suggestion
            counter += 1

        # Log to TensorBoard
        writer.add_scalar("Loss/best", best_loss, epoch)
```

File: climanet/train.py [hunk 1]
```diff
@@ -140,17 +140,17 @@ def train_monthly_model(
...
-                gap = avg_epoch_loss - avg_train_loss
+                gap = avg_val_loss - avg_train_loss
                 print(f"Epoch {epoch}: gap between train and val loss: {gap:.6f}")
 
         # Step scheduler
         scheduler.step(avg_epoch_loss)
 
         # Log to TensorBoard
-        writer.add_scalar("Loss/train", avg_epoch_loss, epoch)
         writer.add_scalar("Loss/best", best_loss, epoch)
```
Comment (SarahAlidoost) [inline climanet/train.py]:
```suggestion
```

**Review by SarahAlidoost — CHANGES_REQUESTED** [review]
@rogerkuou thanks for the implementation 👍 In addition to what I suggested on changed code, I have two more comments about the notebook:
1. You added a plotting code (train vs validation loss) to the notebook as below. Can we have the plotting as a util function? also can we use a better axis labels so later we can put the plots in presentations/publications? Here suggestions: replace "`step`" with "`epoch`", in "`ylabel`" replace "`Loss`" with "`Average loss per epoch (K)`". It would be nice if the unit i.e. `K` can be adjusted based on the input data. 

```python
# Load saved training status with tbparse.SummaryReader
reader = SummaryReader(run_dir)

# plot training and validation loss
plt.figure(figsize=(10, 5))
train_loss = reader.scalars[reader.scalars["tag"] == "Loss/train"]
validation_loss = reader.scalars[reader.scalars["tag"] == "Loss/validation"]
plt.plot(train_loss["step"], train_loss["value"], label="Train Loss")
plt.plot(validation_loss["step"], validation_loss["value"], label="Validation Loss")
plt.xlabel("Step")
plt.ylabel("Loss")
plt.legend()
```

2. in the line below, why `chunk` and `compute()` is needed? I can tun the notebook without them:
```python
monthly_data = xr.open_mfdataset(file_names, chunks={'time': 1, 'lat': 240, 'lon': 240}).compute()
```

**Comment by rogerkuou** [comment]
Thanks @SarahAlidoost, I maked the `plot_loss` function, and removed the chunking and compute (related to our discussion [here](https://github.com/ESMValGroup/ClimaNet/pull/69#discussion_r3602775512)). Can you have another look?

**Review by SarahAlidoost — APPROVED** [review]
@rogerkuou thanks 👍 . Let's merge this.
